create_report_rmd.R

```diff
@@ -0,0 +1,124 @@
+library(tidyverse)
+library(readxl)
+library(stringr)
+library(pander)
+library(knitr)
+# library(flextable)
+# library(RColorBrewer)
+# library(janitor)
+
+##########################
+# Load the survey results
+##########################
+
+if( file.exists("data/data.RData")){
+  load("data/data.RData")
+} else{
+  message("You need the data!")
+}
+
+if( file.exists("misc/Data_dictionary_Survey375147.csv")){
+  dict <- read_csv("misc/Data_dictionary_Survey375147.csv")
+} else{
+  message("You need the data dictionary!")
+}
+
+# utility function 
+get_responses <- function( question_number ){
+  idx <- which( dict$`Question number` == question_number )
+  var_names <- dict$`Variable name`[idx]
+  responses <- data %>% select( all_of(var_names) )
+  if( !is.na(dict$Subquestion[ idx[1] ]) ){
+    # remove square brackets at beginning and end, then rename columns:
+    colnames( responses ) <- str_extract( dict$Subquestion[idx], '(?<=\\[)[^{}]+(?=\\])')
+  } else{
+    colnames( responses ) <- "Response"
+  }
+  return( responses )
+}
+
+idx <- which( is.na( dict$`Question number` )) 
+          # which( grepl('upload', dict$`Type of variable`)),
+          # which ( grepl('filecount', dict$`Question number`) ))
+all_questions <- dict$`Question number`[-idx]
+idx <- which( all_questions == "1" )
+all_questions <- unique( all_questions[ idx:length( all_questions )] )
+n_questions <- length( all_questions )
+    
+write_lines("---
```

Comment:
What is the purpose of these few lines? It seems to work, but I'm wondering (a) if the same thing could be accomplished with `rmarkdown::render()` and (b) if this now requires manual knitting of `report.Rmd` (and if so, it seems like we should have a comment explaining such

Reply:
(a) I should have told you not to worry about this script now -- it's work in progress, separate from the missingness script for Ingrid.
(b) this script creates the RMarkdown file that will then be processed by rmarkdown::render to create reports for each country. It might be easier to explain the rationale for this approach in a call.. and you might have good advice on a more efficient way to do this!
(c) lines 45 and 46 manage the multiple occurrences of some question numbers in the `Question_number` column of the data dictionary -- we just want to print each question once.